config/_common_search_paths.py

```diff
@@ -0,0 +1,50 @@
+import os
+
+def charm_path_search(home):
+	# function that searches for charm_path in locations common across
+	# multiple architectures
+	charm_path = os.getenv('CHARM_HOME')
+
+	if (charm_path is None) and (home is not None):
+		if os.path.isdir(home + '/Charm/charm'):
+			charm_path = home + '/Charm/charm'
```

Comment:
I realize this is mostly moved code and not new, but a safer way to do this might be `os.path.join(home, <dir>)`.